### PR TITLE
fix: move icons to `/priv/static` to aid precompilation + webpack updates

### PR DESCRIPTION
**Pull request by thecristen**
#### Summary of changes

**Asana Ticket:** [Investigate asset compilation bugginess](https://app.asana.com/0/555089885850811/1206658096186582/f)

I've determined the cause of the issue we'd all seen was that the precompilation that was creating the `DotcomWeb.ViewHelpers.svg/1` function clauses relies on every SVG icon already being present in `/priv/static`, but our current handling of assets wouldn't move those icons from `/assets/static` to `/priv/static` until when building the assets for production.

Because the set of icons Dotcom uses hardly ever changes, I felt like it made more sense to just relocate them to `/priv` and leave them there, rather than relying on Webpack copying them over. Most of the changes in this PR are from that move.

I also updated all the Webpack-related dependencies to latest. A few small changes to other files are from that.

---

#### General checks
* [x] **Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?** This is a good practice that can facilitate easier reviews.
* [ ] **Testing.** Do the changes include relevant passing updates to tests? This includes updating screenshots. Preferably tests are run locally to verify that there are no test failures created by these changes, before opening a PR.
* [x] **Tech debt.** Have you checked for tech debt you can address in the area you're working in? This can be a good time to address small issues, or create Asana tickets for larger issues.

### CODE REVIEW

File: assets/css/_events.scss [hunk 1]
```diff
@@ -28,8 +28,7 @@
 
 .m-event-listing {
   @include media-breakpoint-down(xs) {
-    /* stylelint-disable-next-line scss/operator-no-unspaced */
-    margin: 0 -$base-spacing;
+    margin: 0 (-$base-spacing);
```
Comment (thecristen) [inline assets/css/_events.scss:31]: This was flagged by Webpack dev server.

File: assets/webpack.config.base.js [hunk 2]
```diff
@@ -87,13 +89,17 @@ module.exports = {
...
                 {
                   name: "removeAttrs",
                   params: {
-                    "attrs": ["id"]
+                    attrs: ["id"]
                   }
+                },
+                {
+                  name: "sortAttrs",
+                  active: false
```
Comment (thecristen) [inline assets/webpack.config.base.js:102]: This was added due to a changed default in the `svgo-loader` upgrade.

File: assets/webpack.config.base.js [hunk 3]
```diff
@@ -120,7 +126,8 @@ module.exports = {
               postcssOptions: {
                 plugins: [
                   postcssPresetEnv({
-                    autoprefixer: { grid: true }
+                    autoprefixer: { grid: true },
+                    enableClientSidePolyfills: true
```
Comment (thecristen) [inline assets/webpack.config.base.js:130]: This was added due to a changed default in the `postcss-preset-env` upgrade.

File: assets/webpack.config.dev.js [hunk 4]
```diff
@@ -22,17 +22,17 @@ module.exports = env =>
         "Access-Control-Allow-Origin": "*"
       },
       devMiddleware: {
-        writeToDisk: path => /\.css$/.test(path)
+        writeToDisk: true
```
Comment (thecristen) [inline assets/webpack.config.dev.js:25]: This changes makes Webpack write everything to `/priv/static` in dev mode, as opposed to only the CSS

File: upload_assets.sh [hunk 5]
```diff
@@ -17,8 +17,6 @@ aws s3 sync "$STATIC_DIR/js" s3://mbta-dotcom/js --size-only --exclude "*" --inc
 aws s3 sync "$STATIC_DIR/images" s3://mbta-dotcom/images --size-only --exclude "*" --include "*-*" --cache-control=$CACHE_CONTROL
 
 # font content-types need to be specified manually
-aws s3 sync "$STATIC_DIR/fonts" s3://mbta-dotcom/fonts --size-only --exclude "*" --include "*.eot" --cache-control=$CACHE_CONTROL --content-type "application/vnd.ms-fontobject"
-aws s3 sync "$STATIC_DIR/fonts" s3://mbta-dotcom/fonts --size-only --exclude "*" --include "*.svg" --cache-control=$CACHE_CONTROL --content-type "image/svg+xml"
```
Comment (thecristen) [inline upload_assets.sh:21]: We don't have any of these extensions in our `/fonts`

**Comment by anthonyshull** [comment]
@thecristen this doesn't work with docker compose.

**Review by anthonyshull — CHANGES_REQUESTED** [review]
This branch doesn't work with docker compose. Running `docker compose -f deploy/dev.yml up -d` both copies of dotcom fail.

`npm ERR! Missing script: "build"`

**Comment by thecristen** [comment]
> `npm ERR! Missing script: "build"`

This was happening to me too. The only thing that fixed it was nuking everything via `docker system prune -a` 😵 

Anyway, I currently have this working locally with both copies of Dotcom running 🥳

**Review by anthonyshull — APPROVED** [review]

**Pull request merged**